test/unit/org/apache/cassandra/cql3/ast/ExpressionEvaluator.java

```diff
@@ -21,126 +21,152 @@
 import java.math.BigDecimal;
 import java.math.BigInteger;
 import java.nio.ByteBuffer;
-import java.util.Optional;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
 
-import org.apache.cassandra.db.marshal.AbstractType;
+import javax.annotation.Nullable;
 
-import static java.util.Optional.of;
+import org.apache.cassandra.db.marshal.AbstractType;
 
 public class ExpressionEvaluator
 {
-    public static Optional<Object> tryEval(Expression e)
+    @Nullable
+    public static Object eval(Expression e)
```

Comment:
returning `Optional` was just causing confusion, and all code paths did `.get()` anyways.

I also needed the ability to return `null` rather than `empty` which i couldn't do with `Optional`